### ISSUE TITLE
python3 install error: Tk/tkImaging.c:396:5: error: ISO C90 forbids mixed declarations and code

### ISSUE THREAD

**Issue by Fak3** [python-pillow/Pillow#2017]
Hi. I tried to install pillow 3.3.0 with pip3:

    sudo pip3 install pillow

I got an error:
```
Building using 2 processes

gcc -pthread -shared build/temp.linux-x86_64-3.4/_webp.o -L/usr/lib -L/lib64 -L/usr/lib64 -L/usr/local/lib -L/lib -L/usr/lib64 -lwebp -lwebpmux -lwebpdemux -lpython3.4m -o build/lib.linux-x86_64-3.4/PIL/_webp.cpython-34m.so

/usr/lib64/gcc/x86_64-suse-linux/4.8/../../../../x86_64-suse-linux/bin/ld: skipping incompatible /usr/lib/libpthread.so when searching for -lpthread

/usr/lib64/gcc/x86_64-suse-linux/4.8/../../../../x86_64-suse-linux/bin/ld: skipping incompatible /usr/lib/libc.so when searching for -lc

building 'PIL._imagingtk' extension

creating build/temp.linux-x86_64-3.4/Tk

gcc -pthread -Wno-unused-result -Werror=declaration-after-statement -DNDEBUG -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -DOPENSSL_LOAD_CONF -fPIC -ITk -I/usr/include/freetype2 -I/tmp/pip_build_root/pillow/libImaging -I/usr/include -I/usr/local/include -I/usr/include/python3.4m -c _imagingtk.c -o build/temp.linux-x86_64-3.4/_imagingtk.o

gcc -pthread -Wno-unused-result -Werror=declaration-after-statement -DNDEBUG -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -DOPENSSL_LOAD_CONF -fPIC -ITk -I/usr/include/freetype2 -I/tmp/pip_build_root/pillow/libImaging -I/usr/include -I/usr/local/include -I/usr/include/python3.4m -c Tk/tkImaging.c -o build/temp.linux-x86_64-3.4/Tk/tkImaging.o

Tk/tkImaging.c: In function ‘_dfunc’:

Tk/tkImaging.c:396:5: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]

     void *func = dlsym(lib_handle, func_name);

     ^

cc1: some warnings being treated as errors

Building using 2 processes

gcc -pthread -shared build/temp.linux-x86_64-3.4/_imagingtk.o build/temp.linux-x86_64-3.4/Tk/tkImaging.o -L/usr/lib -L/lib64 -L/usr/lib64 -L/usr/local/lib -L/lib -L/usr/lib64 -lpython3.4m -o build/lib.linux-x86_64-3.4/PIL/_imagingtk.cpython-34m.so

gcc: error: build/temp.linux-x86_64-3.4/Tk/tkImaging.o: No such file or directory

error: command 'gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python3 -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/pillow/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-2r5iqubl-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip_build_root/pillow
Storing debug log for failure in /root/.pip/pip.log
```
I am using Python 3.4.1

The error only appears on python3. 

Installation using pip2 succeeded:
```
PIL SETUP SUMMARY
    --------------------------------------------------------------------
    version      Pillow 3.3.0
    platform     linux2 2.7.10 (default, May 24 2015, 14:46:10)
                 [GCC]
    --------------------------------------------------------------------
    --- JPEG support available
    *** OPENJPEG (JPEG2000) support not available
    --- ZLIB (PNG/ZIP) support available
    *** LIBIMAGEQUANT support not available
    --- LIBTIFF support available
    --- FREETYPE2 support available
    --- LITTLECMS2 support available
    --- WEBP support available
    --- WEBPMUX support available
    --------------------------------------------------------------------
```

**Comment by mzbqhbc** [comment]
I also had the this problem and don't know how to solve the problem ,now do you konw how to solve

**Comment by wiredfool** [comment]
I'm surprised that Windows didn't barf on this.  It should be possible to fix by hoisting the declaration to the top of the block, separate from the assignment.

Ping @matthew-brett

**Comment by mzbqhbc** [comment]
I just downloaded the pillow3.20 installation will not have this problem. 
It seems that the problem is 3.30 BUG

**Comment by wiredfool** [comment]
Fixed in #2033
